src/Rx/RxLoader.php

```diff
@@ -8,23 +8,26 @@
 final class RxLoader
 {
 
-    public static function load(string $filename): ?\stdClass
+    public static function load(string $filename): \stdClass
     {
 
         $fileContents = file_get_contents($filename);
         $fileExt = pathinfo($filename, PATHINFO_EXTENSION);
+        $typeOfFile = 'Unknown';
+        $contentsParsed = null;
 
         if (in_array($fileExt, ['yaml', 'yml']) || substr($fileContents, 0, 3) == '---') {
+            $typeOfFile = 'YAML';
             $contentsParsed = Yaml::parse($fileContents, Yaml::PARSE_OBJECT_FOR_MAP);
-        } else {
+        } else if (in_array($fileExt, ['json', 'js'])) {
+            $typeOfFile = 'JSON';
             $contentsParsed = json_decode($fileContents);
-            if (is_null($contentsParsed)) {
-                throw new RxException(sprintf('Unable to parse %s contents in \'%s\'.', in_array($fileExt, ['json', 'js']) ? 'JSON' : 'Unknown', $filename));
-            }
         }
 
+        if (is_null($contentsParsed)) {
```

Comment:
If `Yaml::parse` fails, it looks like a `ParseException` is triggered, so I'm not sure a null check here would suffice. We'd likely need to wrap `Yaml::parse` in its own try/catch and convert the `ParseException` to an `RxException`. See https://symfony.com/doc/current/components/yaml.html#reading-yaml-contents for reference.

Reply:
I'm pretty sure JSON supports the same. I thought you were guarding against reading an empty file as YAML. I've not run this YAML reader. I'll check it out

Reply:
yeah the most compatible for 7.x would be `json_last_error() === JSON_ERROR_NONE` to check. rather than guarding null which may well be the contents of json

http://sandbox.onlinephpfunctions.com/code/0c1bce587f21d6e8da6f7e6e46628f274f706379

Reply:
Interestingly 5.4 thinks most things are not an error ;)